src/mapd-con-es6.js

```diff
@@ -294,45 +299,97 @@ class MapdCon {
     this.queryTimes[queryId] = execution_time_ms
   }
 
-  promisifyThriftMethod = (client, sessionId, methodName, args) =>
+  // ** Event publishing **
```

Comment:
It makes me a little bit nervous to hand-roll our own event emitter implementation. I understand that we need to support both Node and the browser, but have we looked into if there's a common event emitter API between the two we could pull in? Or, failing that, a lightweight library we could use?

Obviously this isn't the most complex thing to implement, but I always like using well-tested APIs when available. It means less stuff we have to worry about with simply, but easy to miss, bugs like null checks and the like,

Reply:
You make a good point - this is re-inventing the wheel and would need further extension and iteration for even some basic features we might want out of pub/sub if we were to start relying on it.

I'll look around for a simple module we could use instead.